notebooks/Notebook3.md

```diff
@@ -2,33 +2,56 @@
+
+To start the week I began by adding in error handling. The user now has a workable interface that allows the user to know when the language that they used is not up to par. If there are parts of their input that are not correct they will receive a message stating that _____ part of their input was wrong and also a provided solution for it. The solution aspect is very similar to what I will provide as documentation. ([link to interface](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/my-dsl/app/src/main/java/my/dsl/Interface/App.java)) ([link to error message files](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/tree/main/my-dsl/app/src/main/java/my/dsl/Interface)) ([link to example error handling](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/my-dsl/app/src/main/java/my/dsl/Parser/StringGrouping.java)).
+
+Next I began working on the implementation of the API. I really wanted to use the google sheets API. After talking with prof Ben I found out that codespaces does not allow API usage to google sheets. However, I should be able to run the project on my own computer through VScode. But this also had its own problems. I had everything set up to test the google sheets api including test code that should have worked, and all of the extra google sheets set up. The set up for google sheets required me to make a google project and set up some keys for it. Avast when running in VScode I did not have maven or gradle or docker set up on my computer. And sadly to get these set up I would need to install brew. After installing brew I found out that I need other requirements on my computer and sadly my computer wont update enough to get the required resources. So after all this set up and testing for proper dependencies I hit a wall that I simply cannot get through. But this sad tail does have a better ending.
+
+After some sleep I figured that there must be another online sheet like resource that uses simple http requests. And after some research light got shed on spreadsheet.com and their open API. Now the easy part of using spreadsheet.com and their api Is that I can send http requests with no hassle. The challenge of using spreadsheet.com is that they require their input of data to be in JSON. Now if you have no background in JSON this gets very confusing. Luckily I do have some background. The challenges that now lay ahead of me are converting my pre-existing data structure into JSON. I spent the rest of the week experimenting with different strategies to do so. I will provide a file that has some of my test code for this. It is very rough and unorganized. But I believe I have found and made some methods that will be able to get the job done. Next week I will plan on finalizing this aspect and making the code and the inputs into the sheet look nice. ([example code not yet in project](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/.devcontainer/Examplecode))
```

Comment:
I liked your description of the issues you were experiencing and the approach you took to start resolving them. I looked through the code linked here and it does look like a lot of effort to try to parse through JSON requests to use the API so I hope it does not require too much more work to get functioning!